### PR TITLE
Make split_on_transmission file thresholds configurable

### PR DESCRIPTION
**Pull request by Snaky-SPB**
The file-splitting thresholds for `split_on_transmission` file outputs were hard-coded constants in `output.cpp`:

- min transmission time: 1.0 s
- max transmission time: 3600 s
- max idle time: 0.5 s

This adds three global configuration keys (in seconds) so they can be tuned without recompiling:

| Key | Default | Meaning |
|-----|---------|---------|
| `min_transmission_time` | 1.0 | the file is not closed on silence until the transmission has lasted at least this long |
| `max_transmission_time` | 3600.0 | the file is closed automatically after this duration, even mid-transmission |
| `max_transmission_idle` | 0.5 | the file is closed after this much time without audio |

## Changes

- `src/helper_functions.{h,cpp}`: the close decision is factored out into a testable `should_close_transmission_file()`
- `src/test_helper_functions.cpp`: unit tests covering the boundary cases (strict comparisons, idle before min duration, max duration)
- `src/rtl_airband.{h,cpp}`: new globals + config parsing with validation; invalid combinations (`min_transmission_time <= 0`, `max_transmission_time <= min_transmission_time`, `max_transmission_idle <= 0`) are rejected with a clear error
- `src/output.cpp`: `close_if_necessary()` uses the configured values instead of static constants
- `CLAUDE.md`: fixed outdated sections on mixers (no `inputs:` section - channels connect via a `type = "mixer"` output) and scan mode (per-frequency settings are parallel lists)

Default behavior is unchanged when the keys are not set.

## Testing

- unit tests: 55/55 pass, NFM and non-NFM debug builds (Alpine, `-Werror`)
- clang-format 14: clean
- manual: binary starts cleanly with the new keys set; invalid values (`max_transmission_time < min_transmission_time`) are rejected with `Configuration error: invalid split_on_transmission settings ...`

Note: the user wiki page about `split_on_transmission` should be updated with the new keys (separate repo).

Example:

```
min_transmission_time = 10.0
max_transmission_idle = 60.0
```

### CODE REVIEW

File: src/rtl_airband.cpp [hunk 1]
```diff
@@ -846,6 +849,16 @@ int main(int argc, char* argv[]) {
             log_scan_activity = true;
         if (root.exists("stats_filepath"))
             stats_filepath = strdup(root["stats_filepath"]);
+        if (root.exists("min_transmission_time"))
+            min_transmission_time = (double)(root["min_transmission_time"]);
+        if (root.exists("max_transmission_time"))
+            max_transmission_time = (double)(root["max_transmission_time"]);
+        if (root.exists("max_transmission_idle"))
+            max_transmission_idle = (double)(root["max_transmission_idle"]);
+        if (min_transmission_time <= 0.0 || max_transmission_time <= min_transmission_time || max_transmission_idle <= 0.0) {
```
Comment (charlie-foxtrot) [inline src/rtl_airband.cpp]: `min_transmission_time` must be >= 1.0 otherwise two files could have the same output filename (timestamp in file name is 1-second resolution)

File: src/rtl_airband.cpp [hunk 2]
```diff
@@ -846,6 +849,16 @@ int main(int argc, char* argv[]) {
             log_scan_activity = true;
         if (root.exists("stats_filepath"))
             stats_filepath = strdup(root["stats_filepath"]);
+        if (root.exists("min_transmission_time"))
+            min_transmission_time = (double)(root["min_transmission_time"]);
```
Comment (charlie-foxtrot) [inline src/rtl_airband.cpp]: reading as a double makes `1.0` work but `1` doesn't, read as a double or int (see others for examples)

**Review by charlie-foxtrot — CHANGES_REQUESTED** [review]
Thanks for the PR!

I added a few comments but nothing major.

Making this globally configurable is great, but if you're up for it, allowing per-channel overrides would be even better (ie per-channel setting overrides global config that overrides default in code)

File: src/rtl_airband.cpp [hunk 3]
```diff
@@ -83,6 +83,9 @@ bool multiple_demod_threads = false;
 bool multiple_output_threads = false;
 bool log_scan_activity = false;
 char* stats_filepath = NULL;
+double min_transmission_time = 1.0;
+double max_transmission_time = 60.0 * 60.0;
+double max_transmission_idle = 0.5;
```
Comment (charlie-foxtrot) [inline src/rtl_airband.cpp]: Looking more closely at what these actually do I think they are mis-named in the current code.  `min_transmission_time` and `max_transmission_time` actually operate on the split file output, not on the transmission.  for example, a transmission that is less than the `min_transmission_time` won't actually be dropped, it will still come out after the `max_transmission_idle` has expired.

Renaming them to `split_min_file_time`, `split_max_file_time`, and `split_max_idle_time` (and the config parameters) may be more clear, but I wonder if this is the functionality you were looking for (vs actually min / max transmission).

It does seem like having `split_max_file_time` configurable would be a good addition, then `split_min_file_time` and `split_max_idle_time` may be less valuable.

There have been other requests for an actual minimum transmission time (ie drop the output if the transmission is shorter than X seconds) but that may be a different PR

File: src/rtl_airband.cpp (hunk 1 above)
Comment (Snaky-SPB) [reply src/rtl_airband.cpp]: Done - validation now requires `split_min_file_time >= 1.0` (both globally and per-output), since the filename timestamp has 1-second resolution.

File: src/rtl_airband.cpp (hunk 2 above)
Comment (Snaky-SPB) [reply src/rtl_airband.cpp]: Done - the values are now read as int or float (a unit-tested `setting_as_double` helper, following the `getType()` pattern used for `gain` and `squelch_snr_threshold`), both globally and per-output.

File: src/rtl_airband.cpp (hunk 3 above)
Comment (Snaky-SPB) [reply src/rtl_airband.cpp]: Agreed - I noticed the same mismatch between the names and the actual behavior: these parameters don't control the transmission itself, they control when the split file is closed. I've renamed them to `split_min_file_time`, `split_max_file_time`, and `split_max_idle_time` (and the helper to `should_close_split_file`) to match what they actually do.

As for a "real" minimum transmission time: ideally `min_transmission_time` should be implemented the way the name suggests - buffer the incoming audio and only start writing to the file once the transmission has lasted at least that long, discarding shorter bursts. That's a bigger behavioral change, so I've kept it out of this PR and would like to pick it up as a follow-up.

**Comment by Snaky-SPB** [comment]
Done - per-output overrides are now supported: a per-output setting overrides the global config, which overrides the default in code. They apply to both `file` and `rawfile` outputs (e.g. a per-channel `split_max_idle_time = 60.0`).

File: src/output.cpp [hunk 4]
```diff
@@ -382,7 +378,7 @@ static void close_if_necessary(output_t* output) {
         double duration_sec = delta_sec(&fdata->open_time, &current_time);
         double idle_sec = delta_sec(&fdata->last_write_time, &current_time);
 
-        if (duration_sec > MAX_TRANSMISSION_TIME_SEC || (duration_sec > MIN_TRANSMISSION_TIME_SEC && idle_sec > MAX_TRANSMISSION_IDLE_SEC)) {
+        if (should_close_split_file(duration_sec, idle_sec, fdata->split_min_file_time, fdata->split_max_file_time, fdata->split_max_idle_time)) {
             debug_print("closing file %s, duration %f sec, idle %f sec\n", fdata->file_path.c_str(), duration_sec, idle_sec);
             close_file(output);
```
Comment (gabeschine) [inline src/output.cpp:383]: What do we think about providing consumers of these files a way to associate multiple output mp3s together as a single transmission? What is great about this PR is that is provides an upper-bound for latency before a file can be acted on/uploaded/listened to. What's unfortunate about the PR is that it loses the concept of "this is actually one transmission" in the sense that the audio that used to be one mp3 file and represented one key down/key up (ish - at least with high probability) is now something that would have to be reconstructed using other means on the consuming side, and with less data to effectively perform the operation.

I see a couple options:

1. Hold the filename's timestamp at the start timestamp of each transmission, and then if the max-time split happens, append `_part1`, then `_part2`, ...
2. Associate some MP3 metadata to all the files so that they are each annotated with the start timestamp of the transmission.

Example for 1:

- `my_channel_20260101_120102_part1.mp3`
- `my_channel_20260101_120102_part2.mp3`
- ...

Example for 2:

- `my_channel_20260101_120102.mp3` (annotated with `transmission_start=20260101_120102`)
- `my_channel_20260101_120112.mp3` (annotated with `transmission_start=20260101_120102`)
- ...

Option 1 pros/cons:
- Pro: a consumer can recognize that a full-transmission mp3 can be reconstructed, and can select the right files, without opening the files.
- Con: requires tracking state so `close_file()` can rename appropriately

Option 2 pros/cons:
- Pro: does not require consumers, who already expect files to follow the existing naming pattern, to do anything differently.
- Con: same as (1)

**Review by gabeschine — COMMENTED** [review]
a drive-by review. hope that's OK - we can discuss the approach, and also whether we want to move ahead in this PR or a follow-up.

i'm personally inclined to move ahead with option (1) due to its listed "pro": when writing a processing pipeline if decisions can be made without opening and reading the file, things are a lot more efficient. but i'm open to being convinced otherwise.

cheers.

**Comment by charlie-foxtrot** [comment]
@Snaky-SPB sorry for the delay in getting this in, I have a few more minor things, but instead of commenting here I'll just push some commits before merging

@gabeschine I'll add something in a follow up PR.  FWIW I'm leaning towards putting it in the MP3 metadata so the output filename format doesn't change.  I also want to think through what happens if something is "stuck open" for multiple hours and creates 1,000s of parts

**Review by charlie-foxtrot — APPROVED** [review]

**Comment by charlie-foxtrot** [comment]
@Snaky-SPB take a look at the commit I added, if you're good with that I'll merge

**Comment by Snaky-SPB** [comment]
Reviewed the commit — all good. It builds cleanly on the existing logic: config keys and the per-output > global > default precedence are unchanged, and the float precision fix plus the improved error reporting are good additions. Unit and system tests are covered and CI is green.

Go ahead and merge.

One more thing: the user wiki page about split_on_transmission (in the separate wiki repo) will need updating with the new config keys — I'll take care of that.
